lib/SymbolGraphGen/SymbolGraph.cpp

```diff
@@ -426,6 +426,17 @@ void SymbolGraph::recordDefaultImplementationRelationships(Symbol S) {
           recordEdge(Symbol(this, VD, nullptr),
                      Symbol(this, MemberVD, nullptr),
                      RelationshipKind::DefaultImplementationOf());
+
+          // If P is from a different module, and it's being added to a type
+          // from the current module, add a `memberOf` relation to the extended
+          // protocol.
+          if (MemberVD->getModuleContext() != &M && VD->getDeclContext()) {
```

Comment:
Is it safe to do a pointer comparison here? Can there be two different pointers to the same module?

Reply:
That's a good point. I'm not sure if it happens in practice, but it's reasonable that there may be two different instances of the same module information. I'll see about comparing some field(s) instead of the pointer itself.